### ISSUE TITLE
Issue with new include/exclude feature

### ISSUE THREAD

**Issue by simast** [pyros2097/rust-embed#150]
Hm, just tried new include/exclude feature:

```rust
#[derive(RustEmbed)]
#[folder = "assets/data/"]
#[include = "**/*.yaml"]
pub struct DataAssets;
````

And my cargo dependency is defined as:

```toml
rust-embed = { version = "6", features = ["include-exclude"] }
```

(lock file shows I have `rust-embed 6.1.0`).

Yet when I call `DataAssets::iter()` I am still getting other files in the list (not just yaml files). Am I missing something, maybe there is a better example somewhere? Thanks.

cc @mbme @pyros2097 @AzureMarker 

p.s. I am using resolver ="2" in Cargo.toml

**Comment by simast** [comment]
Found the issue - for some reason if you have `resolver = "2"` in your Cargo.toml this feature does not work. Note that v2 resolver will be default in Rust edition 2021.

**Comment by AzureMarker** [comment]
I can reproduce this, but only in debug builds. Release builds work correctly. I made a test crate and hooked it up to my local rust-embed project, and added debugging lines to `get_files`:
```
❯ cargo run
   Compiling rust-embed-utils v7.0.0 (/home/mark/Projects/CLionProjects/rust-embed/utils)
   Compiling rust-embed-impl v6.1.0 (/home/mark/Projects/CLionProjects/rust-embed/impl)
   Compiling rust-embed v6.1.0 (/home/mark/Projects/CLionProjects/rust-embed)
   Compiling test-rust-embed v0.1.0 (/home/mark/test-rust-embed)
Starting get_files with includes = ["**/*.yaml"], excludes = []
Considering test.yaml in get_files
is_path_included: include-exclude feature is enabled
  Included
Considering test.txt in get_files
is_path_included: include-exclude feature is enabled
  Not included
    Finished dev [unoptimized + debuginfo] target(s) in 0.76s
     Running `target/debug/test-rust-embed`
Starting get_files with includes = ["**/*.yaml"], excludes = []
Considering test.yaml in get_files
is_path_included: include-exclude feature is disabled
  Included
Considering test.txt in get_files
is_path_included: include-exclude feature is disabled
  Included
["test.yaml", "test.txt"]
```

Something is definitely up. The first output is when the macro embeds the files, the second is when the program calls rust-embed to dynamically load the data. It's compiling the utils crate twice with different features:
```
❯ cargo tree --duplicates
rust-embed-utils v7.0.0 (/home/mark/Projects/CLionProjects/rust-embed/utils)
└── rust-embed v6.1.0 (/home/mark/Projects/CLionProjects/rust-embed)
    └── test-rust-embed v0.1.0 (/home/mark/test-rust-embed)

rust-embed-utils v7.0.0 (/home/mark/Projects/CLionProjects/rust-embed/utils)
└── rust-embed-impl v6.1.0 (proc-macro) (/home/mark/Projects/CLionProjects/rust-embed/impl)
    └── rust-embed v6.1.0 (/home/mark/Projects/CLionProjects/rust-embed) (*)
```

After some research, I found a relevant post on Rust Internals:
https://internals.rust-lang.org/t/concerns-about-making-resolver-2-the-default-in-2021-edition/14684/6

Turns out, our `include-exclude` feature in the rust-embed crate only lists the impl (proc-macro) crate's `include-exclude` feature. Transitively, this enables the util crate's `include-exclude` feature, but with the v2 resolver that means the feature is only enabled when the `impl` crate is used, which is when it's used as a proc macro. When the utils crate is used at runtime to dynamically load the file, it does not have the feature enabled because the `rust-embed` crate didn't list it. The fix is to explicitly enable both the impl and util features in the top-level rust-embed crate. I've opened a PR to do this:

**Comment by pyros2097** [comment]
Fixed in v6.2.0 thanks @AzureMarker .
